cli/command/secret/inspect.go

```diff
@@ -0,0 +1,56 @@
+package secret
+
+import (
+	"context"
+
+	"github.com/docker/docker/cli"
+	"github.com/docker/docker/cli/command"
+	"github.com/docker/docker/cli/command/inspect"
+	"github.com/spf13/cobra"
+)
+
+type inspectOptions struct {
+	name   string
+	format string
+}
+
+func newSecretInspectCommand(dockerCli *command.DockerCli) *cobra.Command {
+	opts := inspectOptions{}
+	cmd := &cobra.Command{
+		Use:   "inspect [name]",
+		Short: "Inspect a secret",
+		Args:  cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.name = args[0]
+			return runSecretInspect(dockerCli, opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.format, "format", "f", "", "Format the output using the given go template")
+	return cmd
+}
+
+func runSecretInspect(dockerCli *command.DockerCli, opts inspectOptions) error {
+	client := dockerCli.Client()
+	ctx := context.Background()
+
+	// attempt to lookup secret by name
+	secrets, err := getSecrets(client, ctx, []string{opts.name})
```

Comment:
it attempts to lookup by name, however if nothing is returned then it assumes the passed was an id and tries to query the api with that.  if that fails, an error is returned to the user.  this is so both `docker secret inspect ssh-key` and `docker secret inspect kgftpkg3448p338ly5ts9i5sf` both work